rars/tools/TimerTool.java

```diff
@@ -0,0 +1,342 @@
+
+Permission is hereby granted, free of charge, to any person obtaining 
+a copy of this software and associated documentation files (the 
+"Software"), to deal in the Software without restriction, including 
+without limitation the rights to use, copy, modify, merge, publish, 
+distribute, sublicense, and/or sell copies of the Software, and to 
+permit persons to whom the Software is furnished to do so, subject 
+to the following conditions:
+
+The above copyright notice and this permission notice shall be 
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, 
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF 
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. 
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR 
+ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION 
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+(MIT license, http://www.opensource.org/licenses/mit-license.html)
+*/
+
+package rars.tools;
+
+import rars.Globals;
+import rars.riscv.hardware.AccessNotice;
+import rars.riscv.hardware.MemoryAccessNotice;
+import rars.riscv.hardware.Memory;
+import rars.riscv.hardware.InterruptController;
+import rars.riscv.hardware.ControlAndStatusRegisterFile;
+import rars.riscv.hardware.AddressErrorException;
+
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.awt.GridLayout;
+import java.awt.FlowLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.*;
+
+
+/**
+ * A RARS tool used to implement a timing module and timer inturrpts.
+ **/
+public class TimerTool extends AbstractToolAndApplication {
+    private static String heading = "Timer Tool";
+    private static String version = "Version 1.0 (Zachary Selk)";
+    private static final int TIME_ADDRESS = Memory.memoryMapBaseAddress + 0x18;
+    private static final int TIME_CMP_ADDRESS = Memory.memoryMapBaseAddress + 0x20;
+
+    // GUI window sections
+    private static JPanel panelTools;
+    private TimePanel timePanel;
+
+    // Internal time values
+    private static long time = 0L;      // The current time of the program (starting from 0)
+    private static long startTime = 0L; // Tmp unix time used to keep track of how much time has passed
+    private static long savedTime = 0L; // Accumulates time as we pause/play the timer
+
+    // Timing threads
+    private static TimeCmpDaemon timeCmp = null; // Watches for changes made to timecmp
+    private Timer timer = new Timer();
+    private Tick tick = new Tick(); // Runs every millisecond to decide if a timer inturrupt should be raised
+
+    // Internal timing flags
+    private static boolean postInterrupt = false; // Signals when timecmp has been writen to
```

Comment:
This line appears to be unused whereas the one in TimeCmpDaemon is used. Did you forget to delete this one?

Reply:
Yes I did.